First draft of test roadmap plan

JamesWTruher opened the pull request:
I'm looking for comments and suggestions

> ready for merge<

JamesWTruher committed:
First draft of test roadmap plan

msftclas commented:
Hi __@JamesWTruher__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <p>
        It looks like you're a Microsoft contributor (Jim Truher). If you're full-time, we DON'T require a Contribution License Agreement. If you are a vendor, please DO sign the electronic Contribution License Agreement. It will take 2 minutes and there's no faxing! https://cla.microsoft.com.
    </p>

TTYL, MSBOT;

adityapatwardhan commented on docs/testing-guidelines/TestRoadmap.md:
```diff
@@ -0,0 +1,193 @@
+The infrastructure used for these tests is STEX which is not an open environment.
+We will need to create automation to create and configure the test systems in the test matrix and then invoke tests on them.
+It is not clear that our current CI systems can accomodate our needs here as neither AppVeyor or Travis can supply us with all of the OS images needed.
+We may need to create our own heterogeneous environment in Azure, or look to other teams (MS Build Lab) for assistance.
```
We should talk to Bryce and find out about the Jenkins system, it may be able to support our needs.

brycem replied:
Abolutely - One nice thing about most of your PowerShell WTT test content that I've noticed is that your test development pattern seems to enforce have a clean(ish) separation of your test code from the test infrastructure.  I've been playing with DSC resources to create single and multi-machine configs ([based on the TLG](http://social.technet.microsoft.com/wiki/contents/articles/1262.test-lab-guides.aspx#Base_Configuration_TLG)) in Azure that are optimized to enable faster and (cheaper) clean\resets between runs.  We should sit down and figure out which STEX templates you're most interested in getting running.

adityapatwardhan commented on docs/testing-guidelines/TestRoadmap.md:
```diff
@@ -0,0 +1,193 @@
+
+There are a number of KPIs which we could report on:
+* Code KPIs
+    * What is the coverage of our `CI` tests
```
>coverage [](start = 18, length = 8)

By coverage, do you mean code coverage blocks/arcs or % of cmdlets / parameters sets or some other metric?

JamesWTruher replied:
i meant % blocks covered - I've updated the language to clarify

adityapatwardhan commented on docs/testing-guidelines/TestRoadmap.md:
```diff
@@ -0,0 +1,193 @@
+* 90% code coverage of language elements (coverage error code paths may not be 100%) 
+* 60% Code Coverage on Windows via Github tests
+* Acceptance by 50% PowerShell MVPs (via Survey)
+* Acceptance by Partners (via Survey)
```
Should we have some metric for cross-remoting scenario?

JamesWTruher committed:
Update document with clarifying language for code coverage

Also add an acceptance criteria for remoting tests
include Jenkins with mention of MS Build Lab for possible test infrastructure
